### PR TITLE
LISA-1583: Add reinstate account endpoint

### PR DESCRIPTION
**Pull request by anand462**

### CODE REVIEW

**Comment by anand462** [comment]
Please check the raml file before merging. Check locally if that is looking good.

File: resources/public/api/conf/1.0/testdata/reinstate-account.md [hunk 1]
```diff
@@ -38,35 +38,60 @@
             </td>
         </tr>
         <tr>
-            <td><p>Request for an account that is open or active</p><p class="code--block">lisaManagerReferenceNumber: <a href="https://test-developer.service.hmrc.gov.uk/api-documentation/docs/api/service/lisa-api/1.0#testing-the-api">Use your test user profile</a><br>accountId: 8765432100</p></td>
+            <td><p>Request for an account that is open or active</p><p class="code--block">lisaManagerReferenceNumber: <a href="https://test-developer.service.hmrc.gov.uk/api-documentation/docs/api/service/lisa-api/1.0#testing-the-api">Use your test user profile</a><br>accountId: 0000000905</p></td>
                 <td><p> empty payload </p>
```
Comment (chriswaugh) [inline resources/public/api/conf/1.0/testdata/reinstate-account.md]: Two things:
1. Missing a closing `</td>`
2. Please just leave it empty, don't specify 'empty payload'

Same thing applies in other areas of this file.

**Comment by chriswaugh** [comment]
Also: we're missing the `INVESTOR_ACCOUNT_ALREADY_OPEN` error scenario

File: resources/public/api/conf/1.0/testdata/reinstate-account.md (hunk 1 above)
Comment (chriswaugh) [reply resources/public/api/conf/1.0/testdata/reinstate-account.md]: (on line 42)

File: app/uk/gov/hmrc/lisaapi/controllers/ReinstateAccountController.scala [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        case ReinstateLisaAccountSuccessResponse => {
+          LisaMetrics.incrementMetrics(startTime, LisaMetricKeys.REINSTATE)
+
+          auditService.audit(
+            auditType = "accountReinstated",
+            path = getReinstateEndpointUrl(lisaManager, accountId),
+            auditData = Map(ZREF -> lisaManager, "accountId" -> accountId)
+          )
+
+          val data = ApiResponseData(message = "The account has been re-instated to a status of open", accountId = Some(accountId))
```
Comment (chriswaugh) [inline app/uk/gov/hmrc/lisaapi/controllers/ReinstateAccountController.scala]: We should be keeping all the API responses together in one place (currently the Responses.scala file)

File: resources/public/api/conf/1.0/application.raml [hunk 3]
```diff
@@ -371,6 +371,87 @@ types:
...
                                   }
+                /reinstate-account:
+                  post:
+                    is: [headers.acceptHeader, headers.contentHeader]
+                    displayName: Reinstate a LISA account
+                    description: Reinstate a LISA account that has been closed.
+                    (annotations.scope): "write:lisa"
+                    securedBy: [ sec.oauth_2_0: { scopes: [ "write:lisa" ] } ]
+                    (annotations.sandboxData): !include testdata/reinstate-account.md
+                    body:
```
Comment (chriswaugh) [inline resources/public/api/conf/1.0/application.raml]: don't think you should specify the body in the RAML if there isn't one.

File: resources/public/api/conf/1.0/application.raml [hunk 4]
```diff
@@ -442,8 +442,8 @@ types:
                                 description: The LISA account is already open.
                                 value: |
                                   {
-                                    "code" : "INVESTOR_ACCOUNT_ALREADY_VOID",
-                                    "message" : "The Investor Account is already void"
+                                    "code" : "INVESTOR_ACCOUNT_ALREADY_OPEN",
+                                    "message" : "The account already has a ststus of Open"
```
Comment (chriswaugh) [inline resources/public/api/conf/1.0/application.raml]: typo on 'status'

Comment (chriswaugh) [reply resources/public/api/conf/1.0/application.raml]: capital O? is that right?

**Comment by chriswaugh** [comment]
LGTM :+1: - some remaining questions on error message & documentation content but we have an upcoming content review to address them

**Pull request merged**